telemetry/specs/INT.mdk

```diff
@@ -1220,6 +1251,128 @@ Metadata if added by the source. The total stack length can be determined
+  If the Domain Specific ID matches any Domain ID known to this node, then additional processing
+  of the Domain Specific Flags and Domain Specific Instruction is required and Domain Specific
+  Metadata is also sent to the monitoring systems.
+
+  Some metadata can be defined as 'source-only' metadata by the INT domain. Those metadata
+  will be sent only by the INT source, not by INT transit or sink devices.
+  In a sense, 'source-only' bits do not serve as instructions for downstream INT devices
+  to follow. In INT-MX mode, the source node should not set the bits corresponding to 
+  "source-only" metadata in the DS Instruction bitmap. However, the source node should ensure
+  that it sets the bits corresponding to "source-only" metadata in DSMdBits bitmap of the 
+  Telemetry Report. This is needed for the monitoring system (or any consumer of the metadata) 
+  to know how to parse and use the data.
+
+  The amount of Domain Specific Metadata sent by each hop must be a multiple of 4 bytes,
+  determined from the Domain Specific Instruction. 
+  
+  If the Domain Specific ID does not match any Domain ID known to this node, then 
+  the transit node is required to either:
+  
+  - Send the metadata corresponding to INT Baseline Instruction, or
+      
+  - Skip INT processing altogether and not send any metadata the monitoring systems.
+
+* Each INT Transit device along the path that supports INT-MX sends its own metadata
+values as specified in the instruction bitmap and DS instructions.
+  - If a device is unable to provide a metadata value specified in the
+    instruction bitmap because its value is not available, it must ensure that the
+    corresponding bit in the "RepMdbit" and/or "DSMdbit" is not set in the Telemetry Report.
+  
+    Details of the metadata semantics and format of the Telemetry Report can 
+    be accessed at the link below:
+    
+    https://github.com/p4lang/p4-applications/blob/master/docs/telemetry_report_v2.0.pdf
+    
+  - Reserved bits in the instruction bitmap are to be handled as described. If an
+    INT transit hop receives a reserved bit set in the instruction bitmap (e.g.
+    set by a INT source that is running a newer version), the transit hop must
+    must ensure that the corresponding reserved bits  "RepMdbit" and/or "DSMdbit" 
+    are not set in the Telemetry Report sent to the monitoring system.
+  
+    Details of the metadata semantics and format of the Telemetry Report can 
+    be accessed at the link below:
+    
+    https://github.com/p4lang/p4-applications/blob/master/docs/telemetry_report_v2.0.pdf
+    
+* Summary of the field usage
+  - The INT Source must set the following fields:
+    - Ver, D, M, and Instruction Bitmap.
+    - INT Source must set all reserved bits to zero.
+    - INT Source may set the Domain-specific fields. 
+  - Intermediate transit devices should not modify the INT-MX header.
```

Comment:
s/should/must

If you want the ability for DS Flags to be modified at transit, then you need to call that out. Domain Specific ID and DS Instruction must not be modified at transit.

Reply:
Added text.